crates/node/p2p/src/net/driver.rs

```diff
@@ -47,12 +52,47 @@ impl Network {
     /// Starts the Discv5 peer discovery & libp2p services
     /// and continually listens for new peers and messages to handle
     pub fn start(mut self) -> Result<(), TransportError<std::io::Error>> {
+        let mut rpc = self.rpc.unwrap_or_else(|| tokio::sync::mpsc::channel(1).1);
         let mut handler = self.discovery.start();
         self.gossip.listen()?;
         let mut interval = tokio::time::interval(tokio::time::Duration::from_secs(5));
         tokio::spawn(async move {
             loop {
                 select! {
+                    req = rpc.recv() => {
```

Comment:
Small nit: We should probably `bias` this `select!`.

By default, this event loop will poll all of the futures in pseudo-random order, but there's a hierarchy here in terms of which futures we want to poll first (i.e., we never want to block unsafe block gossip because of RPC requests.)

not blocking on this PR, this is a problem we'll need to sort out everywhere.

```rs
select! {
    biased;

    // ... futures in order of poll priority ...
}
```